core/src/org/pentaho/di/core/logging/LoggingBuffer.java

```diff
@@ -260,6 +271,31 @@ public void removeChannelFromBuffer( String id ) {
     }
   }
 
+  public void waitSpoonLogTabRefresherBeforeRemoving() {
+    // check environment
+    if ( KettleClientEnvironment.ClientType.SPOON.equals( KettleClientEnvironment.getInstance().getClient() ) ) {
+      // check threads
+      String currentTreadName = Thread.currentThread().getName();
+      if ( Const.JOB_EXECUTOR_DISCARD_LINES_THREAD.equals( currentTreadName )
+        || Const.TRANS_EXECUTOR_DISCARD_LINES_THREAD.equals( currentTreadName ) ) {
+        // set timeout expiration for discard lines waiting
+        long timeoutExpired = System.currentTimeMillis() + Const.KETTLE_DISCARD_LOG_LINES_WAIT_TIMEOUT;
```

Comment:
This seems awkward - why add in System.currentTimeMillis() only to subtract it below ? The method signature is wait( long timeout ) - so why not just use buffer.wait( Const.KETTLE_DISCARD_LOG_LINES_WAIT_TIMEOUT ) ?